include/triton/Dialect/TritonGPU/Transforms/Utility.h

```diff
@@ -255,8 +255,11 @@ namespace mlir::triton {
 /// Replace all uses of `oldUse` with `val` and propagate the type if needed.
 /// This is useful when we need to change a memory descriptor from immutable to
 /// mutable.
-void replaceUsesAndPropagateType(OpBuilder &builder, Operation *oldUse,
-                                 Value val);
+/// The callback is invoked for each pair of an old and a cloned memdesc op
+/// as the type is propagated.
+void replaceUsesAndPropagateType(
+    OpBuilder &builder, Operation *oldUse, Value val,
+    std::function<void(Operation *, Operation *)> callback = nullptr);
```

Comment:
Add the callback param to fix a bug I just found. Since `replaceUsesAndPropagateType` propagates the type by creating new memdesc ops which are not returned to the caller, WarpSchedule doesn't get to see them. When we do `schedule->serialize(loop)` at the end of aref insert, the schedule encounters such unknown memdesc ops and somehow ends up assigning a random partition to them. So `memdesc_trans` op got assigned a partition 2, for example. 

I use the callback to insert the newly-created memdesc ops into the schedule.